packages/core/src/lib/__tests__/radio.spec.ts

```diff
@@ -1,142 +1,108 @@
 import { describe, it, expect, vi } from 'vitest';
-import { render, screen, waitFor } from '@testing-library/svelte';
+import { render, screen } from '@testing-library/svelte';
+import Radio from './radio.spec.svelte';
 import userEvent from '@testing-library/user-event';
-import { Radio } from '$lib';
-import * as MDI from '@mdi/js';
 
 describe('Radio', () => {
+  const common = {
+    name: 'Test radios',
+    options: ['Option 1', 'Option 2', 'Option 3'],
+    selected: 'Option 1',
+  };
+
   it('Renders radio options correctly', () => {
-    render(Radio, { options: ['Option1', 'Option2'], selected: 'Option1' });
-    expect(screen.getByText('Option1')).toBeVisible();
-    expect(screen.getByText('Option2')).toBeVisible();
+    render(Radio, common);
+    expect(screen.getByLabelText('Option 1')).toBeVisible();
+    expect(screen.getByLabelText('Option 2')).toBeVisible();
+    expect(screen.getByLabelText('Option 3')).toBeVisible();
   });
 
   it('Marks the selected option correctly', () => {
-    render(Radio, { options: ['Option1', 'Option2'], selected: 'Option1' });
-    expect(screen.getByText('Option1')).toHaveClass(
-      'bg-light border-gray-6 text-default font-semibold'
-    );
+    render(Radio, common);
+    expect(
+      screen.getByLabelText('Option 1').parentElement?.parentElement
```

Comment:
This `parentElement.parentElement` feels a little brittle / coupled with implementation details. Could this be refactored so it's a little more semantically structured?